src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,78 @@
+                .stream()
+                .filter(maleSelector)
+                .collect(Collectors.toList());
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only people whose age is from `fromAge` and to `maleToAge` (for men)
-     * or to `femaleToAge` (for women) inclusively.
-     * <p>
-     * Example: select people of working age
-     * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
-     */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        Predicate<Person> workablePeopleSelector = human -> human.getAge() >= fromAge
+                && ((human.getSex() == Person.Sex.MAN && human.getAge() <= maleToAge)
+                || (human.getSex() == Person.Sex.WOMAN && human.getAge() <= femaleToAge));
+        return peopleList
+                .stream()
+                .filter(workablePeopleSelector)
+                .collect(Collectors.toList());
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age`, `sex` and `cats` fields,
-     * and each `Cat` having a `name` and `age`),
-     * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
-     */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        Predicate<Person> femaleSelector = human -> human.getSex() == Person.Sex.WOMAN
+                                                && human.getAge() >= femaleAge;
+        return peopleList
+                .stream()
+                .filter(femaleSelector)
+                .map(human -> human.getCats().stream())
+                .flatMap(Stream::distinct)
+                .map(Cat::getName)
+                .collect(Collectors.toList());
     }
 
-    /**
-     * Your help with a election is needed. Given list of candidates, where each element
-     * has Candidate.class type.
-     * Check which candidates are eligible to apply for president position and return their
-     * names sorted alphabetically.
-     * The requirements are: person should be older than 35 years, should be allowed to vote,
-     * have nationality - 'Ukrainian'
-     * and live in Ukraine for 10 years. For the last requirement use field periodsInUkr,
-     * which has following view: "2002-2015"
-     * We want to reuse our validation in future, so let's write our own impl of Predicate
-     * parametrized with Candidate in CandidateValidator.
-     */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        CandidateValidator candidateValidator = new CandidateValidator();
```

Comment:
let's remember about abstraction)
```suggestion
        Predicate<Candidate> candidateValidator = new CandidateValidator();
```